index.js

```diff
@@ -75,23 +82,75 @@ function compression (options) {
 
     // proxy
 
-    res.write = function write (chunk, encoding) {
-      if (ended) {
+    res.write = function write (chunk, encoding, callback) {
+      if (chunk === null) {
+        // throw ERR_STREAM_NULL_VALUES
+        return _write.call(this, chunk, encoding, callback)
+      } else if (typeof chunk === 'string' || typeof chunk.fill === 'function' || isUint8Array(chunk)) {
+        // noop
+      } else {
+        // throw ERR_INVALID_ARG_TYPE
+        return _write.call(this, chunk, encoding, callback)
+      }
+
+      if (!callback && typeof encoding === 'function') {
+        callback = encoding
+        encoding = undefined
+      }
+
+      if (typeof callback !== 'function') {
+        callback = noop
+      }
+
+      if (res.destroyed || res.finished || ended) {
+        // HACK: node doesn't expose internal errors,
+        // we need to fake response to throw underlying errors type
+        var fakeRes = new ServerResponse({})
```

Comment:
I feel like this solution is quite hacky and has a high likelihood of breaking in a future Node.js. we really should try to think on a better solution for this than creating a response object and trying to fake a socket object just to get an error. 🤔

Reply:
Yes, I understand, but the Node.js project seems to be agressive is breaking projects who are "using the api wrong" even though citgm tells them it is broken, at lesst in the https components. Probably should check if this module it is citgm...

Reply:
Looks like this module is not included it citgm, so really cannot use a hack like this without some blessing from node.js... is this needed if we drop some node.js versions?